src/actions/courseActions.js

```diff
@@ -0,0 +1,53 @@
+ import * as types from './actionTypes';
+ import CourseApi from '../api/mockCourseApi';
+ import {beginAjaxCall, ajaxCallError} from './ajaxStatusActions';
+
+export const createCourseSuccess=(course)=>{
+    return{type:types.CREATE_COURSE_SUCCESS, course};
+};
+
+export const loadCoursesSuccess = (courses)=> {
+  return {type: types.LOAD_COURSES_SUCCESS, courses};
+};
+
+export const loadCourses = () => {
+    return function (dispatch ) {
+      dispatch(beginAjaxCall());
+      return CourseApi.getAllCourses().then(courses => {
+        dispatch(loadCoursesSuccess(courses));
+      }).catch(error => {
+        throw(error);
+      });
+    };
+  };
+  export const updateCourseSuccess = (course)=> {
+    return {type: types.UPDATE_COURSE_SUCCESS, course};
+  };
+
+    export const saveCourse = (course)=> {
+      return function (dispatch, getState) {
+        dispatch(beginAjaxCall());
+        return CourseApi.saveCourse(course).then(course => {
+          course.id ? dispatch(updateCourseSuccess(course)) : dispatch(createCourseSuccess(course));
+        }).catch(error => {
+
+          throw(error);
+        });
+      };
+  };
+
+  export const  deleteCourse = (course)=> {
+    return function (dispatch, getState) {
+
```

Comment:
You don't need the second params `getState` since it's not been used inside of the action.